File: app/src/main/java/com/OdiousPanda/rainbow/Adapters/DailyForecastAdapter.java
package com.OdiousPanda.rainbow.Adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.OdiousPanda.rainbow.DataModel.Weather.Daily;
import com.OdiousPanda.rainbow.R;
import com.OdiousPanda.rainbow.Utilities.PreferencesUtil;
import com.OdiousPanda.rainbow.Utilities.UnitConverter;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class DailyForecastAdapter extends RecyclerView.Adapter<DailyForecastAdapter.ForecastViewHolder> {
    private Context context;
    private Daily dailyData;

    public DailyForecastAdapter(Context context, Daily daily) {
        this.context = context;
        this.dailyData = daily;
    }


    @NonNull
    @Override
    public DailyForecastAdapter.ForecastViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.daily_forecast_item, parent, false);
        return new DailyForecastAdapter.ForecastViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull DailyForecastAdapter.ForecastViewHolder holder, int position) {
        String currentTempUnit = PreferencesUtil.getTemperatureUnit(context);
        SimpleDateFormat sdf = new SimpleDateFormat("EEEE", Locale.getDefault());
        Date date = new Date((long) dailyData.getData().get(position).getTime() * 1000);
        String weekday = sdf.format(date);
        float temp = (dailyData.getData().get(position).getTemperatureMax() + dailyData.getData().get(position).getTemperatureMin()) / 2;
        String textTemp = UnitConverter.convertToTemperatureUnitClean(temp, currentTempUnit);
        String iconName = dailyData.getData().get(position).getIcon().replace("-", "_");
        int iconResourceId = context.getResources().getIdentifier("drawable/" + iconName + "_b", null, context.getPackageName());
        holder.setData(weekday, textTemp, iconResourceId);
    }

    @Override
    public int getItemCount() {
        return dailyData.getData().size() - 1;
    }

    class ForecastViewHolder extends RecyclerView.ViewHolder {
        private TextView date;
        private ImageView icon;
        private TextView temp;

        ForecastViewHolder(@NonNull View itemView) {
            super(itemView);
            date = itemView.findViewById(R.id.date_forecast);
            icon = itemView.findViewById(R.id.icon_forecast);
            temp = itemView.findViewById(R.id.temp_forecast);
        }

        void setData(String date, String temp, int iconResource) {
            this.date.setText(date);
            this.temp.setText(temp);
            this.icon.setImageResource(iconResource);
        }
    }
}

File: app/src/main/java/com/odiousPanda/rainbowKt/adapters/DailyForecastAdapter.kt
package com.odiousPanda.rainbowKt.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.odiousPanda.rainbowKt.databinding.DailyForecastItemBinding
import com.odiousPanda.rainbowKt.model.dataSource.weather.Daily
import com.odiousPanda.rainbowKt.utilities.PreferencesUtil
import com.odiousPanda.rainbowKt.utilities.UnitConverter
import java.text.SimpleDateFormat
import java.util.*

class DailyForecastAdapter(private val context: Context, private val daily: Daily) :
    RecyclerView.Adapter<DailyForecastAdapter.DailyForecastViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DailyForecastViewHolder {
        val binding =
            DailyForecastItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return DailyForecastViewHolder(binding)
    }

    override fun getItemCount(): Int {
        return daily.data.size
    }

    override fun onBindViewHolder(holder: DailyForecastViewHolder, position: Int) {
        val currentTempUnit = PreferencesUtil.getTemperatureUnit(context).toString()
        val sdf = SimpleDateFormat("EEEE", Locale.getDefault())
        val weekDay = sdf.format(daily.data[position].time * 1000)
        val temp = (daily.data[position].temperatureMax + daily.data[position].temperatureMin) / 2
        val tempString =
            UnitConverter.convertToTemperatureUnitClean(temp, currentTempUnit).toString()
        val iconName = daily.data[position].icon.replace("-", "_")
        val iconRes =
            context.resources.getIdentifier("drawable/${iconName}_b", null, context.packageName)
        holder.setData(weekDay, tempString, iconRes)
    }

    class DailyForecastViewHolder(private val binding: DailyForecastItemBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun setData(date: String, temp: String, iconRes: Int) {
            binding.dateForecast.text = date
            binding.tempForecast.text = temp
            binding.iconForecast.setImageResource(iconRes)
        }
    }
}